Nos/Models/CoreData/Event+Hydration.swift

```diff
@@ -179,30 +179,37 @@ extension Event {
         }
     }
     
-    private func hydrateMuteList(from jsonEvent: JSONEvent, context: NSManagedObjectContext) {
-        let mutedKeys = jsonEvent.tags.map { $0[1] }
+    private func hydrateMuteList(from jsonEvent: JSONEvent, context: NSManagedObjectContext) throws {
+        guard createdAt! > author?.lastUpdatedMuteList ?? Date.distantPast else {
+            return
+        }
+        
+        author?.lastUpdatedMuteList = Date(timeIntervalSince1970: TimeInterval(jsonEvent.createdAt))
+        
+        let mutedKeys = Set(jsonEvent.tags.map { $0[1] })
         
         let request = Author.allAuthorsRequest(muted: true)
         
         // Un-Mute anyone (locally only) who is muted but not in the mutedKeys
         if let authors = try? context.fetch(request) {
             for author in authors where !mutedKeys.contains(author.hexadecimalPublicKey!) {
                 author.muted = false
```

Comment:
@mplorentz , why are we unmuting anyone who is muted but not in the muted key? Is it possible to be muted but the key shouldn't be in the muted keys list?

Reply:
That makes sense.